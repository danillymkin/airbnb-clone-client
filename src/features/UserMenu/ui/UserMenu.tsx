import React, { FunctionComponent } from 'react';
import { AiOutlineMenu } from 'react-icons/ai';
import { Menu, MenuItem } from '@/shared/ui/Menu';
import { Avatar } from '@/shared/ui/Avatar';

export const UserMenu: FunctionComponent = (): JSX.Element => {
  return (
    <div className={'relative'} data-testid={'user-menu'}>
      <div className={'flex items-center gap-3'}>
        <div
          className={
            'hidden cursor-pointer rounded-full px-4 py-3 text-sm font-semibold transition hover:bg-neutral-100 md:block'
          }
        >
          Airbnb your home
        </div>

        <Menu
          trigger={
            <div
              className={
                'flex cursor-pointer items-center gap-3 rounded-full border border-neutral-200 p-4 transition hover:shadow-md md:px-2 md:py-1'
              }
            >
              <AiOutlineMenu />

              <div className={'hidden md:block'}>
                <Avatar />
              </div>
            </div>
          }
        >
          <MenuItem>Sign up</MenuItem>

          <MenuItem>Login</MenuItem>
        </Menu>
      </div>
    </div>
  );
};
