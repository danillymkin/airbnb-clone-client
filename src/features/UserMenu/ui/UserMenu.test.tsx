import { render, screen } from '@testing-library/react';
import { UserMenu } from '@/features/UserMenu';

describe('User Menu', () => {
  it('should render UserMenu', () => {
    render(<UserMenu />);
    const userMenu = screen.getByTestId('user-menu');
    expect(userMenu).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    render(<UserMenu />);
    const userMenu = screen.getByTestId('user-menu');
    expect(userMenu).toMatchSnapshot();
  });

  it('should render your home button', () => {
    render(<UserMenu />);
    const yourHomeBtn = screen.getByText(/airbnb your home/i);
    expect(yourHomeBtn).toBeInTheDocument();
  });

  it('should render menu', () => {
    render(<UserMenu />);
    const menu = screen.getByTestId('menu');
    expect(menu).toBeInTheDocument();
  });
});
