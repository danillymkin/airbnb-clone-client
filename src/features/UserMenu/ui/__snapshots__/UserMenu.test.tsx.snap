// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Menu should match snapshot 1`] = `
<div
  class="relative"
  data-testid="user-menu"
>
  <div
    class="flex items-center gap-3"
  >
    <div
      class="hidden cursor-pointer rounded-full px-4 py-3 text-sm font-semibold transition hover:bg-neutral-100 md:block"
    >
      Airbnb your home
    </div>
    <div
      class="relative"
      data-headlessui-state=""
      data-testid="menu"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-headlessui-state=""
        id="headlessui-menu-button-:r2:"
        type="button"
      >
        <div
          class="flex cursor-pointer items-center gap-3 rounded-full border border-neutral-200 p-4 transition hover:shadow-md md:px-2 md:py-1"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
          <div
            class="hidden md:block"
          >
            <img
              alt="Avatar"
              class="rounded-full"
              data-nimg="1"
              decoding="async"
              height="30"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=32&q=75 1x, /_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="30"
            />
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
